server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -40,7 +40,7 @@
  * This class holds all {@link DiscoveryNode} in the cluster and provides convenience methods to
  * access, modify merge / diff discovery nodes.
  */
-public class DiscoveryNodes extends AbstractCollection<DiscoveryNode> implements SimpleDiffable<DiscoveryNodes> {
+public class DiscoveryNodes implements Streamable<DiscoveryNode>, SimpleDiffable<DiscoveryNodes> {
```

Comment:
`DiscoveryNodes` is a good example here.

We are not using it with collections add/addAll/remove inherited from `AbstractCollection`, however there are valid stream/iterator usages of this class